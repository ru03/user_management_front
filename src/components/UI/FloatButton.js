import React from 'react';
import { Fab } from '@material-ui/core';

const FloatButton = ({ children, color = 'primary', clickHandler }) => {
  return (
    <Fab color={color} onClick={() => clickHandler && clickHandler()}>
      {children}
    </Fab>
  )
}

export default FloatButton;