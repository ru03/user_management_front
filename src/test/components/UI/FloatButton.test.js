import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import FloatButton from '../../../components/UI/FloatButton';

describe('FloatButton Component', () => {
  it('mounts', () => {
    const { asFragment } = render(<FloatButton children='Submit' />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('has a title', () => {
    const { getByText } = render(<FloatButton children='Submit' />);
    expect(getByText('Submit')).toHaveTextContent('Submit');
  });

  it('has called on click handler', () => {
    const onClickHandler = jest.fn();
    const { container } = render(<FloatButton children='Submit' clickHandler={onClickHandler} />);
    fireEvent.click(container.firstChild);
    expect(onClickHandler).toHaveBeenCalledTimes(1);
  });
});
