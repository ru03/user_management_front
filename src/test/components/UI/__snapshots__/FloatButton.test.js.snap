// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloatButton Component mounts 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiFab-root MuiFab-primary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiFab-label"
    >
      Submit
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
